internal/driver/controllerserver.go

```diff
@@ -268,7 +268,10 @@ func (linodeCS *LinodeControllerServer) ControllerPublishVolume(ctx context.Cont
 	if canAttach, err := linodeCS.canAttach(ctx, instance); err != nil {
 		return &csi.ControllerPublishVolumeResponse{}, status.Error(codes.Internal, err.Error())
 	} else if !canAttach {
-		limit := linodeCS.maxVolumeAttachments(instance)
+		limit, err := linodeCS.maxVolumeAttachments(ctx, instance)
+		if err != nil {
+			return &csi.ControllerPublishVolumeResponse{}, status.Error(codes.ResourceExhausted, "max number of volumes already attached to instance")
```

Comment:
not sure this error is right (e.g. instance being nil or an api error listing the instance disks)

Reply:
Ah, good catch! Thank you. I will handle those errors separately.